From e771f0c1b41f0b0bcf9cc4ff40e4f74b8dd9cab8 Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Wed, 12 Jan 2022 09:46:56 -0600
Subject: [PATCH 1/1] ARM:dts:aspeed: Add Quartz dts adc121c sensors

adc121c adc voltage sensors are added with Quartz device tree

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts | 56 ++++++++++++++++++++-
 1 file changed, 54 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index 50e0f1c493ef..fb1298882405 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -55,6 +55,17 @@
 		};
 	};

+	iio-hwmon-adc121c {
+		compatible = "iio-hwmon";
+		oemname0 = "FPGA_VDD_CORE";
+		oemname1 = "FPGA_VDD_18";
+		oemname2 = "FPGA_VDD_33";
+		oemname3 = "LOM_VDD_33";
+		oemname4 = "LOM_VDD_12";
+		oemname5 = "VDD_12_DUAL";
+		io-channels = <&adc0 0>, <&adc1 0>, <&adc2 0>, <&adc3 0>,
+					<&adc4 0>, <&adc5 0>;
+	};
 };

 &mdio0 {
@@ -75,8 +86,6 @@
 	pinctrl-0 = <&pinctrl_rgmii4_default>;
 };

-
-
 &fmc {
 	status = "okay";
 	fmc-spi-user-mode;
@@ -520,6 +529,49 @@
 				reg = <0x4f>;
 			};
 		};
+
+		i2c@6 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <6>;
+
+			adc0: fpga_vdd_core@50 {
+				compatible = "ti,adc121c";
+				reg = <0x50>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_core";
+			};
+			adc1: fpga_vdd_18@51 {
+				compatible = "ti,adc121c";
+				reg = <0x51>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_18";
+			};
+			adc2: fpga_vdd_33@52 {
+				compatible = "ti,adc121c";
+				reg = <0x52>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_33";
+			};
+			adc3: lom_vdd_33@54 {
+				compatible = "ti,adc121c";
+				reg = <0x54>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_33";
+			};
+			adc4: lom_vdd_12@55 {
+				compatible = "ti,adc121c";
+				reg = <0x55>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_12";
+			};
+			adc5: vdd_12_dual@56 {
+				compatible = "ti,adc121c";
+				reg = <0x56>;
+				#io-channel-cells = <1>;
+				label = "vdd_12_dual";
+			};
+		};
 	};
 };

--
2.17.1

